indexing-service/src/main/java/org/apache/druid/indexing/scheduledbatch/ScheduledBatchTaskManager.java

```diff
@@ -0,0 +1,342 @@
+      {
+        if (taskStatus.isComplete()) {
+          statusTracker.onTaskCompleted(taskId, taskStatus);
+        }
+      }
+    };
+  }
+
+  /**
+   * Initializes and starts scheduled ingestion for the specified {@code supervisorId}, scheduling tasks
+   * using the provided {@code spec} and {@code cronSchedulerConfig}.
+   */
+  public void startScheduledIngestion(
+      final String supervisorId,
+      final String dataSource,
+      final CronSchedulerConfig cronSchedulerConfig,
+      final ClientSqlQuery spec
+  )
+  {
+    log.info(
+        "Starting scheduled batch ingestion into datasource[%s] with supervisorId[%s], cronSchedule[%s] and spec[%s].",
+        supervisorId, dataSource, cronSchedulerConfig, spec
+    );
+    final ScheduledBatchTask taskScheduler = new ScheduledBatchTask(supervisorId, dataSource, cronSchedulerConfig, spec);
+    taskScheduler.startScheduling();
+    supervisorToTaskScheduler.put(supervisorId, taskScheduler);
+  }
+
+  /**
+   * Stops the scheduler for the specified {@code supervisorId}, pausing task submissions
+   * while retaining the ability to track the supervisor's status.
+   */
+  public void stopScheduledIngestion(final String supervisorId)
+  {
+    log.info("Stopping scheduled batch ingestion for supervisorId[%s].", supervisorId);
+    final ScheduledBatchTask taskScheduler = supervisorToTaskScheduler.get(supervisorId);
+    if (taskScheduler != null) {
+      // Don't remove the supervisorId from supervisorToManager. We want to be able to track the status
+      // for suspended supervisors to track any in-flight tasks. stop() will clean up all state completely.
+      taskScheduler.stopScheduling();
+    }
+  }
+
+  /**
+   * Starts the scheduled batch task manager by registering the {@link TaskRunnerListener}.
+   * This allows tracking of any tasks submitted by the batch supervisor.
+   * <p>
+   * Should be invoked when the Overlord service starts or during leadership transitions.
+   * </p>
+   */
+  public void start()
+  {
+    log.info("Starting scheduled batch task manager.");
+    final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
+    if (taskRunnerOptional.isPresent()) {
+      taskRunnerOptional.get().registerListener(taskRunnerListener, Execs.directExecutor());
+    } else {
+      log.warn("Task runner not registered for scheduled batch task manager.");
+    }
+  }
+
+  /**
+   * Stops the scheduled batch task manager by unregistering the previously registered {@link TaskRunnerListener}.
+   * <p>
+   * Should be invoked when the Overlord service stops or during leadership transitions.
+   * </p>
+   */
+  public void stop()
+  {
+    log.info("Stopping scheduled batch task manager.");
+    final Optional<TaskRunner> taskRunnerOptional = taskMaster.getTaskRunner();
+    if (taskRunnerOptional.isPresent()) {
+      taskRunnerOptional.get().unregisterListener(taskRunnerListener.getListenerId());
+    }
+
+    supervisorToTaskScheduler.clear();
```

Comment:
Before doing a clear, we should also tearDown each of the `ScheduledBatchTask` objects.